### ISSUE TITLE
Qualitative results

### ISSUE THREAD

**Issue by micaban** [amazon-science/page-link-path-based-gnn-explanation#3]
Hi! 
I first ran the PaGE-Link to explain trained GNN models with aug_citation. Then I executed the eval_explanations and got the quantitative results. How could I get some examples of paths explanations as shown in the qualitative results?

**Comment by ShichangZh** [comment]
Thank you for your question. We basically used the [plot_hetero_graph](https://github.com/amazon-science/page-link-path-based-gnn-explanation/blob/6de8b2d956c9c53f00833340987ea2b107300525/utils.py#L894C5-L894C5) function to generate those results. Maybe you can check the function first? I will try to provide an example notebook asap.

**Comment by micaban** [comment]
Thanks for your answer!
I checked the function and tried to implement it but could not arrived to the plot, I always get some error.
As I understood from the code, sending a DGL model to plot_hetero_graph would be the minimum to execute it.

Tried to load the saved models and the explanations with the following lines of code:
model = torch.load('saved_explanations/pagelink_aug_citation_model_pred_edge_to_comp_g_edge_mask',map_location=torch.device('cpu'))

model = model.HeteroRGCNLayer('aug_citation',128,128,128)
model.load_state_dict(torch.load("saved_models/aug_citation_model.pth"),map_location=torch.device('cpu'))

Got the following error when calling the plot function:
utils.plot_hetero_graph(model)
AttributeError: 'collections.OrderedDict' object has no attribute 'ntypes'

It's not clear for me what model should I use or if maybe I should made any transformation to get a DGL graph.
If you have any example of loading the model and calling the plot function will be great!


In the other hand, I can't understand the usability of the files that were saved in saved_explanation.
When I loaded to a notebook I execute the following:
```
file = open('saved_explanations/pagelink_aug_citation_model_pred_edge_to_comp_g_edge_mask', 'rb')

# dump information to that file
data = pickle.load(file)
print(len(data))
# close the file
file.close()

print('Showing the pickled data:')

cnt = 0
for item in data:
    print('The data ', cnt, ' is : ', item)
    cnt += 1
```

The first part of the output is the following;
```
412
Showing the pickled data:
The data  0  is :  (('author', 2215), ('paper', 1107))
The data  1  is :  (('author', 2902), ('paper', 578))
The data  2  is :  (('author', 351), ('paper', 528))
The data  3  is :  (('author', 3210), ('paper', 128))
The data  4  is :  (('author', 5513), ('paper', 257))
The data  5  is :  (('author', 690), ('paper', 400))
The data  6  is :  (('author', 759), ('paper', 1555))
The data  7  is :  (('author', 1043), ('paper', 1978))
The data  8  is :  (('author', 488), ('paper', 1834))
The data  9  is :  (('author', 374), ('paper', 1096))
The data  10  is :  (('author', 617), ('paper', 707))
The data  11  is :  (('author', 1887), ('paper', 13))
The data  12  is :  (('author', 1682), ('paper', 2749))
The data  13  is :  (('author', 509), ('paper', 1337))
The data  14  is :  (('author', 3696), ('paper', 2016))
The data  15  is :  (('author', 2309), ('paper', 2414))
The data  16  is :  (('author', 783), ('paper', 210))
```

But I don't know how to access data to understand the explainability. I would like to see some examples of why a link prediction is predicted. I'm accessing the file in a wrong way?

Thanks!

**Comment by ElioShyti** [comment]
Hi @micaban, i think that the input required by the plot_hetero_graph function should be a dgl graph or subgraph not the state_dict of the model.
Also for visualizing the explanation paths i think that you should use the 'pagelink_aug_citation_model_pred_edge_to_paths' file since the other file is the mask.
In 'pagelink_aug_citation_model_pred_edge_to_paths' you can find a dict which keys are the predictions of the 'likes' edge and the values are the explanation paths generated. 

That's how i interpreted the files, I hope I'm right

**Comment by micaban** [comment]
@ElioShyti thanks! I loaded the pagelink_aug_citation_model_pred_edge_to_paths and I was able to see the dictionary with the paths that explain the prediction. I will try to convert the state_dict of the the saved model to a dgl graph to use the graph function, will see if it's possible

**Comment by ShichangZh** [comment]
@micaban Sorry for the late update. @ElioShyti Thank you for helping with the question.

I have added a new notebook `plotting_example.ipynb` for an example of how to use the plot function in utils.py. 

The first half of the notebook is just copy-pasted from eval_explanations.py to load the data and model. Afterwards, I picked one example from the test graph, load the stored explanation mask, and run pagelink.get_paths to extract the paths for plotting. I also updated the `plot_hetero_graph` function in utils.py a little bit, because I realized the previous version wasn't the most recent version and didn't include edge labelling. Sorry for the confusion.

In summary, please refer to the `plotting_example.ipynb` for the details, and let me know if there are any further questions.
